File: packages/scanner/tests/fixtures/python-langchain/agent.py
from langchain_openai import ChatOpenAI
from langchain.agents import AgentExecutor, create_react_agent
from langchain_core.prompts import ChatPromptTemplate
from langchain_community.tools import DuckDuckGoSearchRun

# Initialize the LLM
llm = ChatOpenAI(model="gpt-4o", temperature=0)

# Define tools
search = DuckDuckGoSearchRun()
tools = [search]

# Create the agent
prompt = ChatPromptTemplate.from_messages([
    ("system", "You are a helpful research assistant."),
    ("human", "{input}"),
])
agent = create_react_agent(llm, tools, prompt)
agent_executor = AgentExecutor(agent=agent, tools=tools, verbose=True)

File: packages/scanner/tests/fixtures/ts-langchain/agent.ts
import { ChatOpenAI } from "@langchain/openai";
import { AgentExecutor, createReactAgent } from "langchain/agents";
import { pull } from "langchain/hub";
import { DuckDuckGoSearch } from "@langchain/community/tools/duckduckgo";

const llm = new ChatOpenAI({
  model: "gpt-4o",
  temperature: 0,
});

const tools = [new DuckDuckGoSearch()];

async function main() {
  const prompt = await pull("hwchase17/react");
  const agent = await createReactAgent({ llm, tools, prompt });
  const agentExecutor = new AgentExecutor({ agent, tools });

  const result = await agentExecutor.invoke({
    input: "What are the latest AI agent frameworks?",
  });

  console.log(result);
}

main();
